CalculatorIdo/Script.js

```diff
@@ -0,0 +1,149 @@
+window.onload = () => {clearCalculator();}
+
+var currAns = '';
+var isResultShown = false;
+var isbracketOpen = false;
+
+clearCalculator = () => {document.calcForm.input.value = '';}
+
+del = () => {
+    if(isResultShown) {
+        isResultShown = false;
+        clearCalculator();
+    }
+
+    let currString = document.calcForm.input.value;
+
+    // Deleted the whole word if its a word
+    if(/^[a-zA-Z]+$/.test(currString.charAt(currString.length - 1))) {
+        document.calcForm.input.value = currString.substring(0, currString.length - 1);
+        currString = document.calcForm.input.value;
+
+        while((/^[a-zA-Z]+$/.test(currString.charAt(currString.length - 1)))) {
+            document.calcForm.input.value = currString.substring(0, currString.length - 1);
+            currString = document.calcForm.input.value;
+        }
+    } else {
+        document.calcForm.input.value = currString.substring(0, currString.length - 1);
+    }
+}
+
+insert = (input) => { 
+    if(isResultShown) {
+        isResultShown = false;
+        clearCalculator();
+    }
+
+    document.calcForm.input.value = document.calcForm.input.value + input;
+}
+
+equal = () => {
+    if(isResultShown) {
+        isResultShown = false;
+        clearCalculator();
+    }
+
+    let resultString = document.calcForm.input.value;
+    if(resultString) {
+        let calculatedResult = calculateResult(stringResultToArray(resultString));
+        if(calculatedResult === undefined || isNaN(calculatedResult)) {
+            alert('Syntax error');
+        } else {
+            document.calcForm.input.value = calculatedResult;
+            currAns = calculatedResult;
+            isResultShown = true;
+            isDotAvailable = true;
+        }
+    }
+}
+
+stringResultToArray = (resultString) => {
+    let resultArray = [];
+    let currResult = '';
+    let currChar = '';
+
+    for (let i = 0; i < resultString.length; i++) {
+        currChar = resultString.charAt(i);
+        
+        if (/^[)(*/+-]$/.test(currChar)) {
+            if (currResult === '' && currChar === '-' &&
+                !(i > 0 && resultString.charAt(i-1) === ')')) {
+                currResult = '-';
+            } else {
+                if(currChar === '(' ||
+                   (/^[)*/+-]$/.test(currChar) && currResult === '')) {
+                    resultArray.push(currChar);
+                } else {
+                    resultArray.push(parseFloat(currResult), currChar);
+                }
+                currResult = '';
+            }
+        } else {
+            currResult += currChar;
+        }
+    }
+
+    if (currResult !== '') {
+        resultArray.push(parseFloat(currResult));
+    }
+
+    return resultArray;
+}
+
+calculateResult = (resultArray) => {
+    let operators = [{'*': (a, b) => a * b,
```

Comment:
It should be a const and not let. The easiest way to know if it's a let or a const is start with const until you need to change it's value, if so then it's let